lab_1_classify_by_unigrams/main.py

```diff
@@ -135,6 +211,29 @@ def preprocess_profile(profile: dict) -> dict[str, str | dict] | None:
     In case of corrupt input arguments or lack of keys 'name', 'n_words' and
     'freq' in arguments, None is returned
     """
+    if (not isinstance(profile, dict)
+            or not all(key in profile for key in ('name', 'freq', 'n_words'))
+            or next((token for token in profile['freq'].keys()
+                     if isinstance(token, str)), None) is None):
+        return None
+    freq_dic = {}
+    for token in profile['freq']:
+        if len(token) == 1:
+            freq_dic[token] = profile['freq'][token]
+
+    unigrams_list = list(freq_dic)
+    preprocessed_profile = {'name': profile['name'], 'freq': {}}
+    for unigram in unigrams_list:
+        if not unigram.isupper():
```

Comment:
in profiles there are the same letters both in lower and uppercase. at first i check if the unigram is.upper() so that there are no mistakes while calculating frequencies
why not islower()? because there are also unigrams like a hieroglyph and a symbol indicating the power of the number which don't have a lowercase version